OPHD/XmlSerializer.h

```diff
@@ -1,9 +1,49 @@
 #pragma once
 
+#include <NAS2D/StringUtils.h>
 #include <NAS2D/Xml/Xml.h>
 #include <string>
+#include <unordered_map>
+#include <typeinfo>
+#include <type_traits>
+#include <stdexcept>
 
 
 // Open an xml document
 // Throws a runtime_error if the xml is ill formed or the root element name is incorrect
 NAS2D::Xml::XmlDocument openXmlFile(std::string filename, std::string rootElementName);
+
+int elementToInt(const NAS2D::Xml::XmlElement* element);
+
+template<typename T>
+T stringToEnum(const std::unordered_map<std::string, T>& table, std::string value)
+{
+	auto it = table.find(NAS2D::toLowercase(value));
```

Comment:
There's a bit of an implicit assumption here that key values for the `table` will always be all lowercase. The behavior here can potentially be quite confusing if that's not the case, as lookup would always fail, even if the `value` exactly matched a key.

I would lean towards making this case sensitive. Less confusion if someone uses a table with keys that have non-lowercase letters. Plus, if we're dealing with machine processed data, the case will always be exact, and we would catch inconsistencies more quickly.